Doc/library/stdtypes.rst

```diff
@@ -163,13 +163,12 @@ This table summarizes the comparison operations:
    pair: object; numeric
    pair: objects; comparing
 
-Objects of different types, except different numeric types, never compare equal.
-Furthermore, some types (for example, function objects) support only a degenerate
-notion of comparison where any two objects of that type are unequal.  The ``<``,
-``<=``, ``>`` and ``>=`` operators will raise a :exc:`TypeError` exception when
-comparing a complex number with another built-in numeric type, when the objects
-are of different types that cannot be compared, or in other cases where there is
-no defined ordering.
+Objects of different types, except different numeric types, should never compare equal.
+The ``==`` defaults to :keyword:`is` if both object's :meth:`__eq__` method returns NotImplemented.
```

Comment:
I'd use `class objects` instead of `function objects` -- the intro to this section (L15) mentions those explicitly.

FWIW reading that section makes me think we should replace `compared` on L23 with `compared for equality`.